@here/harp-mapview/lib/VisibleTileSet.ts

```diff
@@ -91,11 +93,25 @@ const MB_FACTOR = 1.0 / (1024.0 * 1024.0);
  * name, but implementation could be improved to omit this limitation.
  */
 class DataSourceCache {
-    static getKey(tileCode: number, dataSource: DataSource): string {
-        return `${dataSource.name}_${tileCode}`;
+    /**
+     * Creates unique tile key for caching based on morton code, tile offset and its data source.
+     * @param mortonCode The tile morton code.
+     * @param offset The tile offset.
+     * @param dataSource The [[DataSource]] from which tile was loaded.
+     */
+    static getKey(mortonCode: number, offset: number, dataSource: DataSource): TileCacheId {
+        return `${dataSource.name}_${mortonCode}_${offset}`;
```

Comment:
have you done a performance test on this? Is it quick to have strings as keys?

Reply:
Here you have:
http://jsben.ch/ac9c9
Map with number keys is 35% faster then with strings, but do you have any idea how we could replace string while still preserving DataSource information to identify each tile within all data sources.